utils.go

```diff
@@ -696,43 +696,22 @@ func (ci *CborInt) UnmarshalCBOR(r io.Reader) error {
 type CborTime time.Time
 
 func (ct *CborTime) MarshalCBOR(w io.Writer) error {
-	b, err := (*time.Time)(ct).MarshalBinary()
-	if err != nil {
-		return err
-	}
+	nsecs := ct.Time().UnixNano()
```

Comment:
nano-nit: this will panic if we try to serialize something with a null `*time.Time`. It should probably check `ct == null` and emit `null` in that case.